src/mtev_http.c

```diff
@@ -317,6 +328,35 @@ struct bchain *bchain_from_data(const void *d, size_t size) {
   return n;
 }
 
+static mtev_http_compress 
+request_compression_type(mtev_http_request *req)
+{
+  const char *content_encoding = NULL;
+  if (req == NULL || req->freed) return MTEV_HTTP_COMPRESS_NONE;
+  
+  mtev_hash_table *headers = mtev_http_request_headers_table(req);
+  mtev_hash_retr_str(headers, "content-encoding", strlen("content-encoding"), 
+                     &content_encoding);
+
+  if (content_encoding == NULL) {
+    return MTEV_HTTP_COMPRESS_NONE;
+  }
+  
+  /* there is no official mime-type for LZ4 so use this for now */
+  int lz4f = strcmp("lz4f", content_encoding);
```

Comment:
I thought, if there's no "official" mime-type, that's when you'd use an `x-` prefix? `x-lz4f`?